src/main/java/org/sonarlint/intellij/issue/IssueProcessor.java

```diff
@@ -47,26 +50,37 @@
   private final DaemonCodeAnalyzer codeAnalyzer;
   private final IssueStore store;
   private final SonarLintConsole console;
+  private final ServerIssueUpdater serverIssueUpdater;
 
-  public IssueProcessor(Project project, IssueMatcher matcher, DaemonCodeAnalyzer codeAnalyzer, IssueStore store) {
+  public IssueProcessor(Project project, IssueMatcher matcher, DaemonCodeAnalyzer codeAnalyzer, IssueStore store, ServerIssueUpdater serverIssueUpdater) {
     super(project);
     this.matcher = matcher;
     this.codeAnalyzer = codeAnalyzer;
     this.store = store;
     this.console = SonarLintConsole.get(project);
+    this.serverIssueUpdater = serverIssueUpdater;
   }
 
-  public void process(final SonarLintJobManager.SonarLintJob job, final Collection<Issue> issues, Collection<ClientInputFile> failedAnalysisFiles) {
-    Map<VirtualFile, Collection<IssuePointer>> map;
+  public void process(final SonarLintJobManager.SonarLintJob job, final Collection<Issue> issues, Collection<ClientInputFile> failedAnalysisFiles, TriggerType trigger) {
+    Map<VirtualFile, Collection<LocalIssuePointer>> map;
     long start = System.currentTimeMillis();
     AccessToken token = ReadAction.start();
     try {
       map = transformIssues(issues, job.files(), failedAnalysisFiles);
+
+      Set<VirtualFile> notAnalyzed =
+        job.files().stream().filter(store::containsFile).collect(Collectors.toSet());
```

Comment:
why not filter first by shouldTrackServerIssues? Would reduce accesses to the store and processing it again after the store.store(map).